dlc/src/util.rs

```diff
@@ -0,0 +1,67 @@
+//! Utility functions not uniquely related to DLC
+
+use bitcoin::util::bip143::SighashComponents;
+use bitcoin::{PrivateKey, Script, SigHashType, Transaction};
+use secp256k1::{Message, Secp256k1, Signature, Signing};
+
+pub(crate) fn get_sig_hash_msg(
```

Comment:
Added some comments I hope it helps.

Reply:
Comment certainly helps, though I think it might even be nice to link out to like https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki or whatever spec this follows